livedata/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -173,22 +173,38 @@ class EventHandlerImpl(
                     event.totalUnreadCount?.let { domainImpl.setTotalUnreadCount(it) }
                     users.putAll(event.message.users().associateBy(User::id))
                     messages[event.message.id] = MessageEntity(event.message)
+                    channelMap[event.cid]?.let {
+                        it.addMessage(MessageEntity(event.message))
+                        channels[it.cid] = it
+                    }
                 }
                 is MessageDeletedEvent -> {
                     event.message.cid = event.cid
                     users.putAll(event.message.users().associateBy(User::id))
                     messages[event.message.id] = MessageEntity(event.message)
+                    channelMap[event.cid]?.let {
+                        it.addMessage(MessageEntity(event.message))
+                        channels[it.cid] = it
+                    }
                 }
                 is MessageUpdatedEvent -> {
                     event.message.cid = event.cid
                     users.putAll(event.message.users().associateBy(User::id))
                     messages[event.message.id] = MessageEntity(event.message)
+                    channelMap[event.cid]?.let {
+                        it.addMessage(MessageEntity(event.message))
+                        channels[it.cid] = it
+                    }
                 }
                 is NotificationMessageNewEvent -> {
                     event.message.cid = event.cid
                     event.totalUnreadCount?.let { domainImpl.setTotalUnreadCount(it) }
                     users.putAll(event.message.users().associateBy(User::id))
                     messages[event.message.id] = MessageEntity(event.message)
+                    channelMap[event.cid]?.let {
```

Comment:
You could perhaps extract this code to a function, it is being repeated 4 times.

Reply:
The same for some of the previous lines in the different branches here

Reply:
The same for some of the previous lines in the different branches here

Reply:
Named this `addMessageData`, but happy to hear more meaningful suggestions